### PR TITLE
Update GET /api/blocks?height=x and GET /api/blocks/:id endpoints - Closes #5249

### LINKED ISSUE #5249: Update GET /api/blocks?height=x and GET /api/blocks/:id endpoints
### Description

Add the API endpoints:

- ` GET /api/blocks?height=x`
- `GET /api/blocks/:id`

according to the following specification:

- id should be in base64 format
- Both endpoints should return a deserialized full block
- Validate if height is number if not `400` should be returned
- Validate if id is base64 formatted string if not `400` should be returned
- Validate `/blocks` endpoint has height query if not `400` should be returned
- Use actions to obtain the data from the framework

### Motivation

The HTTP API should return blocks by height or id.

### Acceptance Criteria

- Block id input must be base64
- Add functional tests
- Register plugin to core, and check if it returns decoded data with height
- Register plugin to core, and check if it returns decoded data with id
- Check if invalid height returns 400 with error message
- Check if invalid id returns 400 with error message
- Check if returns 404 with error message if the block does not exist

### Additional Information

- Check if `/api/blocks` returns 400 with error message

Related issues #5213

### PR DESCRIPTION
**Pull request by pablitovicente**
### What was the problem?

This PR resolves #5249

### How was it solved?

By adding http plugin route `/api/blocks/?height=<block height>`
By adding http plugin route `/api/blocks/<block id>`

### How was it tested?

- Functional tests added

### CODE REVIEW

File: framework-plugins/lisk-framework-http-api-plugin/test/functional/blocks.spec.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+// import { codec } from '@liskhq/lisk-codec';
```
Comment (ManuGowda) [inline framework-plugins/lisk-framework-http-api-plugin/test/functional/blocks.spec.ts:14]: remove if not required?

File: framework-plugins/lisk-framework-http-api-plugin/test/functional/blocks.spec.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+			expect(result.data).toHaveProperty('data.header.timestamp');
+			expect(result.data).toHaveProperty('data.header.height');
+			expect(result.data).toHaveProperty('data.header.previousBlockID');
+			expect(result.data).toHaveProperty('data.header.transactionRoot');
+			expect(result.data).toHaveProperty('data.header.generatorPublicKey');
+			expect(result.data).toHaveProperty('data.header.reward');
+			expect(result.data).toHaveProperty('data.header.asset');
+			expect(result.data).toHaveProperty('data.header.asset.maxHeightPreviouslyForged');
+			expect(result.data).toHaveProperty('data.header.asset.maxHeightPrevoted');
+			expect(result.data).toHaveProperty('data.header.asset.seedReveal');
```
Comment (ManuGowda) [inline framework-plugins/lisk-framework-http-api-plugin/test/functional/blocks.spec.ts:56]: Just a suggestion.
```
expect(Object.keys(result.data)).to.equal(['header', 'payload'])
expect(Object.keys(result.data.header)).to.equal(['id', 'version']) so on 
expect(Object.keys(result.data.header.asset)).to.equal(['maxHeightPreviouslyForged']) so on
```

File: framework-plugins/lisk-framework-http-api-plugin/test/functional/blocks.spec.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
```
Comment (ManuGowda) [inline framework-plugins/lisk-framework-http-api-plugin/test/functional/blocks.spec.ts:117]: if you do not like the above suggestion, then wrap the expectation to function and something like, blockValidation(result). and reuse the same above.

File: framework-plugins/lisk-framework-http-api-plugin/test/functional/blocks.spec.ts (hunk 2 above)
Comment (pablitovicente) [reply framework-plugins/lisk-framework-http-api-plugin/test/functional/blocks.spec.ts]: Changed to this.

File: framework-plugins/lisk-framework-http-api-plugin/test/functional/blocks.spec.ts (hunk 3 above)
Comment (pablitovicente) [reply framework-plugins/lisk-framework-http-api-plugin/test/functional/blocks.spec.ts]: Followed the other suggestion as right now are similar but maybe in the future it changes. Also I prefer not to have tests with much helpers that hide what's going on too much.

**Pull request merged**